### PR TITLE
Add documentation for error E0208

### PR DESCRIPTION
**Pull request by onlineSoftwareDevOK**
Related to https://github.com/rust-lang/rust/issues/61137

### CODE REVIEW

File: compiler/rustc_error_codes/src/error_codes/E0208.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#![feature(rustc_attrs)]
+
+#[rustc_variance]
+struct Test2<'a, 'b, 'c> { // ERROR [-, -, -]
+     x: &'a isize,
+     y: &'b [isize],
+     c: &'c str
+}
+
+fn main() {}
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0208.md:15]: It shouldn't be needed.

File: compiler/rustc_error_codes/src/error_codes/E0208.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Below find a table mapping the symbol shown to the variance:
+
+| **Symbol** | **Variance**  |
+|------------|---------------|
+| -          | contravariant |
+| +          | covariant     |
+| o          | invariant     |
+| *          | bivariant     |
+
+More info at [Rustonomicon] (https://doc.rust-lang.org/nomicon/subtyping.html)
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0208.md:27]: Should be:

```suggestion
More info at [Rustonomicon](https://doc.rust-lang.org/nomicon/subtyping.html).
```

File: compiler/rustc_error_codes/src/error_codes/E0208.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+struct Test2<'a, 'b, 'c> { // ERROR [-, -, -]
+     x: &'a isize,
+     y: &'b [isize],
+     c: &'c str
+}
+
+fn main() {}
+```
+
+Below find a table mapping the symbol shown to the variance:
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0208.md:18]: There is really nothing more to be explained about this error code?

Comment (onlineSoftwareDevOK) [reply compiler/rustc_error_codes/src/error_codes/E0208.md:18]: Thanks for your feedback. What else should be explained about this error code?

File: compiler/rustc_error_codes/src/error_codes/E0208.md (hunk 2 above)
Comment (onlineSoftwareDevOK) [reply compiler/rustc_error_codes/src/error_codes/E0208.md]: Done

File: compiler/rustc_error_codes/src/error_codes/E0208.md (hunk 1 above)
Comment (onlineSoftwareDevOK) [reply compiler/rustc_error_codes/src/error_codes/E0208.md]: Removed

File: compiler/rustc_error_codes/src/error_codes/E0208.md (hunk 3 above)
Comment (GuillaumeGomez) [reply compiler/rustc_error_codes/src/error_codes/E0208.md:17]: Try to adopt this point of view: you don't know what this error code is about at all. What information does the reader need to understand why they got this error?

In the current case, you only show a table of variance, but that's far from enough to understand what's going on in here.

Comment (onlineSoftwareDevOK) [reply compiler/rustc_error_codes/src/error_codes/E0208.md:17]: Thanks. A link to the following page would be enough? https://doc.rust-lang.org/nomicon/subtyping.html

Comment (onlineSoftwareDevOK) [reply compiler/rustc_error_codes/src/error_codes/E0208.md:17]: I have just realized it's already there

Comment (onlineSoftwareDevOK) [reply compiler/rustc_error_codes/src/error_codes/E0208.md:17]: I felt the link to the documentation was enough @mbartlett21 What else can I add here?

**Comment by onlineSoftwareDevOK** [comment]
> The job **`x86_64-gnu-llvm-12`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/6846132042?check_suite_focus=true) [(plain)](https://github.com/rust-lang/rust/commit/557221ef3e7b01e76ea5328c986ca3d798d60a02/checks/6846132042/logs)
> 
> Click to see the possible cause of the failure (guessed by this bot)

@GuillaumeGomez Can you explain why this failed, please? A similar sentence ("For more information...") was added in the following PR but it worked there https://github.com/rust-lang/rust/pull/94449

**Review by compiler-errors — CHANGES_REQUESTED** [review]
I don't think that an internal error needs error documentation. 

The feature gate specifically says "the `#[rustc_error]` attribute is just used for rustc unit tests and will never be stable", and I don't think people working within the compiler need this documentation.

**Comment by onlineSoftwareDevOK** [comment]
> I don't think that an internal error needs error documentation.
> 
> The feature gate specifically says "the `#[rustc_error]` attribute is just used for rustc unit tests and will never be stable", and I don't think people working within the compiler need this documentation.

I prepared this PR because this error code is mentioned here https://github.com/rust-lang/rust/issues/61137#issue-448347560

**Comment by compiler-errors** [comment]
@GuillaumeGomez, perhaps that one should also be marked off the list as unneeded?

**Comment by GuillaumeGomez** [comment]
I made the list based on the undocumented error codes but didn't make difference with internal ones. We can just add a comment like `// internal error code` behind it and not add a long error explanation. Same for the errors listed by @compiler-errors.

**Comment by onlineSoftwareDevOK** [comment]
> I made the list based on the undocumented error codes but didn't make difference with internal ones. We can just add a comment like `// internal error code` behind it and not add a long error explanation. Same for the errors listed by @compiler-errors.

Inside `compiler/rustc_error_codes/src/error_codes.rs`, right?

**Comment by GuillaumeGomez** [comment]
Yes. Something like this:

```rust
E0208, // internal error code
```

**Comment by onlineSoftwareDevOK** [comment]
> Yes. Something like this:
> 
> ```rust
> E0208, // internal error code
> ```

Done

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**